Signal/src/ViewControllers/NewContactThreadViewController.m

```diff
@@ -388,76 +541,107 @@ - (void)updateTableContents
         }
+
+        NSString *text = [NSString stringWithFormat:NSLocalizedString(@"SEND_INVITE_VIA_SMS_BUTTON_FORMAT",
+                                                        @"Text for button to send a Signal invite via SMS. %@ is "
+                                                        @"placeholder for the receipient's phone number."),
+                                   displayName];
+        [inviteeSection addItem:[OWSTableItem disclosureItemWithText:text
+                                                     customRowHeight:self.actionCellHeight
+                                                         actionBlock:^{
+                                                             [weakSelf sendTextToPhoneNumber:phoneNumber.toE164];
+                                                         }]];
+    }
+    if (invitees.count > 0) {
+        [sections addObject:inviteeSection];
     }
 
-    if (hasSearchText && !hasSearchResults) {
-        // No Search Results
 
-        [section addItem:[OWSTableItem softCenterLabelItemWithText:
-                                           NSLocalizedString(@"SETTINGS_BLOCK_LIST_NO_SEARCH_RESULTS",
-                                               @"A label that indicates the user's search has no matching results.")
-                                                   customRowHeight:kActionCellHeight]];
+    if (!hasSearchResults) {
+        // No Search Results
+        OWSTableSection *noResultsSection = [OWSTableSection new];
+        [noResultsSection
+            addItem:[OWSTableItem softCenterLabelItemWithText:
+                                      NSLocalizedString(@"SETTINGS_BLOCK_LIST_NO_SEARCH_RESULTS",
+                                          @"A label that indicates the user's search has no matching results.")
+                                              customRowHeight:self.actionCellHeight]];
+
+        [sections addObject:noResultsSection];
     }
 
-    [contents addSection:section];
-
-    self.tableViewController.contents = contents;
+    return [sections copy];
 }
 
 - (NSArray<SignalAccount *> *)filteredSignalAccounts
 {
-    NSString *searchString = [self.searchBar text];
+    NSString *searchString = self.searchBar.text;
 
     ContactsViewHelper *helper = self.contactsViewHelper;
     return [helper signalAccountsMatchingSearchString:searchString];
 }
 
+- (NSArray<TSGroupThread *> *)filteredGroupThreads
+{
+    NSMutableArray<TSGroupThread *> *matchingThreads = [NSMutableArray new];
+    NSString *searchString = [[self.searchBar.text lowercaseString] ows_stripped];
+    [TSGroupThread enumerateCollectionObjectsUsingBlock:^(id obj, BOOL *stop) {
+        if (![obj isKindOfClass:[TSGroupThread class]]) {
+            // group and contact threads are in the same collection.
+            return;
+        }
+        TSGroupThread *groupThread = (TSGroupThread *)obj;
+        if ([[[groupThread.groupModel.groupName lowercaseString] ows_stripped] containsString:searchString]) {
```

Comment:
I think it'd be nice if our matching algorithm supported "splits", e.g. searching for "Chen M" matched "Matthew Chen".  See the search logic in `ContactsViewHelper`.  In fact, I'd love to move this logic into `ContactsViewHelper`, because eventually I'm hoping to do search in the home view that matches (group name/contact name/contact phone number/non-phone identifier/message content!).

Reply:
I've extracted the group search logic, and it now supports splits. Test probably illustrates this best: https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/test/util/SearcherTest.swift

I also decided to match against group member names as well as the group name. So, for example, searching for "Alice Bob" will list groups to which both Alice and Bob are members.